document/CoreFormat.mdk

```diff
@@ -1120,11 +1185,78 @@ C++: `Namespace::Class::Method(int, double) const &&`
 C#: `Namespace1.Namespace2.Class.Method(System.String, int[])`
 ~
 
+If the `run.logicalLocations` property (&sect;[#run-logicalLocations]) is present,
+the value of the `fullyQualifiedLogicalName` property should be equal to the name of one of the properties on the `run.logicalLocations` object,
+with one exception, described in &sect;[#location-logicalLocationKey].
+
 ~ IsoNote
 The intent of this property is to allow a result log viewer to display the logical location
```

Comment:
Sometimes you don't have physical location data, end of story. Logical location details can help fill the gap in those cases. fullyQualifiedLogicalName exists because we want producers to create it, not consumers (if you require consumers to do it, they will make their own formatting/delimiter decisions). This property is intended for display and suppression (and it helps humans with navigation because they recognize the item in question without special parsing). The additional logical location components assist consumers in organizing data and can help viewers and other non-human processors in potentially navigating to items (where otherwise, they'd have to parse a fullyQualifiedLogicalName correctly).

---
In reply to: [60402106](https://github.com/sarif-standard/sarif-spec/pull/161#discussion_r60402106) [](ancestors = 60402106)